src/temporal/t.rast.aggregate.seasons/t.rast.aggregate.seasons.py

```diff
@@ -0,0 +1,219 @@
+
+    try:
+        vals = options["years"].split("-")
+        years = range(vals)
+    except:
+        try:
+            years = options["years"].split(",")
+        except:
+            if strds.find("@") >= 0:
+                id_ = strds
+            else:
+                id_ = strds + "@" + gscript.gisenv()["MAPSET"]
+            dataset = tgis.dataset_factory("strds", id_)
+            dataset.select(dbif)
+            ext = dataset.get_temporal_extent()
+            years = range(ext.start_time.year, ext.end_time.year)
+
+    method = options["method"]
+    basename = options["basename"]
+    nprocs = int(options["nprocs"])
+    register_null = flags["n"]
+
+    if output_name:
+        out_strds = output_name
+    else:
+        out_strds = basename
+
+    seasons_name = ["spring", "summer", "autumn", "winter"]
+
+    # create new space time vector datasets one for each year to be used as sampler
+    for year in years:
+        season_vect = []
+        for seas in seasons_name:
+            name = f"sample_{seas}_{year}"
+            vect = VectorTopo(name)
+            vect.open("w")
+            point = Point(0, 0)
+            vect.write(point, cat=1)
+            vect.close()
+            map_layer = tgis.space_time_datasets.VectorDataset(f"{name}@{mapset}")
+            if seas == "spring":
+                extent = tgis.AbsoluteTemporalExtent(
+                    start_time=datetime(int(year), 3, 20),
+                    end_time=datetime(int(year), 6, 21),
+                )
+            elif seas == "summer":
+                extent = tgis.AbsoluteTemporalExtent(
+                    start_time=datetime(int(year), 6, 21),
+                    end_time=datetime(int(year), 9, 20),
+                )
+            elif seas == "autumn":
+                extent = tgis.AbsoluteTemporalExtent(
+                    start_time=datetime(int(year), 9, 20),
+                    end_time=datetime(int(year), 12, 21),
+                )
+            elif seas == "winter":
+                extent = tgis.AbsoluteTemporalExtent(
+                    start_time=datetime(int(year), 12, 21),
+                    end_time=datetime(int(year) + 1, 3, 20),
+                )
+            map_layer.set_temporal_extent(extent=extent)
+            season_vect.append(map_layer)
+        outsp = tgis.open_new_stds(
+            f"sample_seasons_{year}",
+            "stvds",
+            "absolute",
+            f"Season vector year {year}",
+            f"Season vector for the year {year}",
+            "mean",
+            dbif,
+            gscript.overwrite(),
+        )
+        tgis.register_map_object_list(
+            "vector",
+            season_vect,
+            outsp,
+            False,
+            None,
+            dbif,
+        )
+
+    process_queue = pymod.ParallelModuleQueue(int(nprocs))
+
+    # create t.rast.aggregate.ds module to be copied
+    mod = pymod.Module("t.rast.aggregate.ds")
```

Comment:
Documentation says:

> Setting [run_](https://grass.osgeo.org/grass78/manuals/libpython/pygrass.modules.interface.html#id9) to False is important, otherwise a parallel processing is not possible